int16 id
float32 positionX
float32 positionY
float32 velocityX
float32 velocityY
---
string msg